### PR TITLE
Listing of KungFuu Finance

### PR DESCRIPTION
**Pull request by kungfuu-finance**
**Twitter Link**: https://twitter.com/kungfuu_fi

**List of audit links if any:**

**Website Link:**
https://www.kungfuu.app/

**Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):**
https://raw.githubusercontent.com/kungfuu-finance/logo/main/kungfuu-dark.svg
https://raw.githubusercontent.com/kungfuu-finance/logo/main/kungfuu-light.svg

**Current TVL:**

**Chain:**
Fantom

**Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)**

**Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)**

**Short Description (to be shown on DefiLlama):**
KungFuu Finance is an auto-staking protocol on Fantom with a 50,000% fixed APY. Owners of Fork It, the leading DeFi fork listing ecosystem

**Token address and ticker if any:**
0x89b61Ab033584918103698953870F07D6db412A3 $KNGFUU

**Category (full list at https://defillama.com/categories) *Please choose only one:**
16. Staking

**Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):**

**forkedFrom (Does your project originate from another project):**
Titano Finance

**methodology (what is being counted as tvl, how is tvl being calculated):**
TVL Includes LP, and a split RFV consisting of USDC in reserve and deployed stablecoin yield farming to fund additional buyback and burn events.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/KungFuu-Finance exports TVL: 
        
 
 
    fantom                    4.51 k
    fantom-staking            0
    staking                   0
    
    total                    4.51 k

File: projects/KungFuu-Finance/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    
+    // Edit TVL of object to be the cumulative tvl
+    const tvl_object = tvl_per_treasury[0]
+    const tvls = tvl_per_treasury.map(o => o[chain === "ftm"?"fantom":chain].tvl)
+    tvl_object[chain === "ftm"?"fantom":chain].tvl = sdk.util.sumChainTvls(tvls)
+    return tvl_object
+}
+
+const kngfuu_token = "0x89b61Ab033584918103698953870F07D6db412A3"
+const treasuries = ["0x914adbe1a641f1e4a8ce22776d342d4c2669f030", "0xaaef45e31e2d2865a4722c1591ba4cd8f6e83bad", "0x0aad04f96b29eb5d31aa52317e19b78a0ffecb8d" ]
```
Comment (g1nt0ki) [inline projects/KungFuu-Finance/index.js:18]: two of the three addresses here are EOA, sorry we dont accept that for tvl

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

**Comment by kungfuu-finance** [comment]
Updated, everything is now under the multi-sig contracts.

**Comment by llamatester** [comment]
The adapter at projects/KungFuu-Finance exports TVL: 
        
 
 
    fantom                    4.32 k
    fantom-staking            0
    staking                   0
    
    total                    4.32 k

File: projects/KungFuu-Finance/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const sdk = require("@defillama/sdk")
```
Comment (g1nt0ki) [inline projects/KungFuu-Finance/index.js:1]: Change to multisig is good, can you update code to below, logic is the same but bit easier to read:

    const { sumTokens } = require('../helper/unwrapLPs')
    const { getChainTransform } = require('../helper/portedTokens')
    const sdk = require("@defillama/sdk")

    const wFTM = '0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83'
    const kngfuu_token = "0x89b61Ab033584918103698953870F07D6db412A3".toLowerCase()

    async function tvl(ts, _block, { fantom: block }) {
      const chain = 'fantom'
      const balances = {}
      const transform = await getChainTransform(chain)
      const owners = ["0x29187b10a04B269Cf067AE013B3ab58d4affaC03", "0xaaef45e31e2d2865a4722c1591ba4cd8f6e83bad", "0xfeAFAF0610fe2c73bB6345080056066aE109B31F"]
      const treasuryTokens = [
        wFTM, // WFTM
        "0x04068da6c83afcfa0e13ba15a6696662335d5b75", // USDC
        "0xf704f5ac5edf152168e07e6f5f108366911250ac", // WFTM/KNGFUU, needs to account only half of it
      ]

      const tokensAndOwners = []
      treasuryTokens.forEach(t => owners.forEach(o => tokensAndOwners.push([t, o])))
      await sumTokens(balances, tokensAndOwners, block, chain, transform, { resolveLP: true })
      delete balances[transform(kngfuu_token)]
      const  { output: results } = await sdk.api.eth.getBalances({ targets: owners, chain, block})
      for (const res of results) sdk.util.sumSingleBalance(balances, transform(wFTM), res.balance)
      return balances
    }

    module.exports = {
      fantom: {
        tvl
      }
    }

**Comment by llamatester** [comment]
The adapter at projects/KungFuu-Finance exports TVL: 
        
 
 
    fantom                    5.03 k
    
    total                    5.03 k

**Comment by kungfuu-finance** [comment]
Twitter Link: https://twitter.com/kungfuu_fi

List of audit links if any:

Website Link:
https://www.kungfuu.finance/

Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
https://raw.githubusercontent.com/kungfuu-finance/logo/main/kungfuu-dark.svg
https://raw.githubusercontent.com/kungfuu-finance/logo/main/kungfuu-light.svg

Current TVL:

Chain:
Fantom

Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)

Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

Short Description (to be shown on DefiLlama):
KungFuu Finance is building a complete DeFi ecosystem, with profit sharing and a deflationary utility token.

Token address and ticker if any:
0x89b61Ab033584918103698953870F07D6db412A3 $KNGFUU

*Category (full list at https://defillama.com/categories) Please choose only one:
16. Staking

Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):

forkedFrom (Does your project originate from another project):
Titano Finance

methodology (what is being counted as tvl, how is tvl being calculated):
TVL Includes LP, and a split RFV consisting of USDC in reserve and deployed stablecoin yield farming to fund additional buyback and burn events.

**Pull request merged**